modal/functions.py

```diff
@@ -679,6 +679,12 @@ def _deps(only_explicit_mounts=False) -> List[_Object]:
                 deps.append(nfs)
             for _, vol in validated_volumes:
                 deps.append(vol)
+
+            # Add implicit dependencies from the function's code
+            objs: list[Object] = get_referred_objects(info.raw_f)
+            _objs: list[_Object] = synchronizer._translate_in(objs)
+            deps += _objs
```

Comment:
do these need to be deduped anywhere?

Reply:
it's fine, as long as the list is the same locally and in the container